src/mlpack/methods/lsh_model/lshmodel.hpp

```diff
@@ -0,0 +1,373 @@
+    * @param kValues The rank of the neighbors used for the statistic, for
+    *     example k = 5 means Ek is the arithmetic mean of the 5th-nearest
+    *     neighbor for different sample sizes.
+    * @param Ek The arithmetic mean of the squared distances of a point and its
+    *      k-nearest neighbor. One column per k.
+    * @param Gk The geometric mean of the squared distances of a point and its
+    *      k-nearest neighbor. One column per k.
+    */
+   void ApproximateKNNStatistics(const arma::Col<size_t>& referenceSizes, 
+                                 const arma::Col<size_t>& kValues,
+                                 const arma::mat& Ek, 
+                                 const arma::mat& Gk);
+
+   /**
+    * This is a helper class that uses the function a * k^b * N^c for some
+    * parameters a, b, c that have been fit to either predict the arithmetic or
+    * geometric mean of the squared distance of a point to its k-nearest
+    * neighbor, given some dataset size N and its k-nearest neighbor.
+    */
+   class DistanceStatisticPredictor
+   {
+    public:
+      //! Empty constructor.
+      DistanceStatisticPredictor() { };
+
+      /** 
+       * Function to construct with training set.
+       *
+       * @param inputSize A vector of input sizes. The first input variable of 
+       *     the regression.
+       * @param kValues A vector of k values. The second input variable of the
+       *     regression.
+       * @param statistic A vector of responses - the value of the statistic for
+       *     each given inputSize.
+       */
+      DistanceStatisticPredictor(const arma::Col<size_t>& inputSize, 
+                                 const arma::Col<size_t>& kValues,
+                                 const arma::mat& statistic) 
+      { Train(inputSize, kValues, statistic); };
+      
+      //! Default destructor.
+      ~DistanceStatisticPredictor() { };
+
+      /**
+       * Function that fits the alpha, beta and gamma parameters.
+       *
+       * @param inputSize A vector of input sizes. The first input variable of 
+       *     the regression.
+       * @param kValues A vector of k values. The second input variable of the
+       *     regression.
+       * @param statistic A vector of responses - the value of the statistic for
+       *     each given inputSize.
+       */
+      double Train(const arma::Col<size_t>& inputSize, 
+                 const arma::Col<size_t>& kValues,
+                 const arma::mat& statistic);
+
+      /** 
+       * Evaluate the statistic for a given dataset size.
+       *
+       * @param N - a new input size for which to evaluate the expected
+       *     statistic.
+       */
+      double Predict(size_t N, size_t k) 
+      { return alpha * std::pow(k, beta) * std::pow(N, gamma); };
+
+      //! Set the alpha parameter.
+      void Alpha(double a) { alpha = a; };
+
+      //! Get the alpha parameter.
+      double Alpha(void) { return alpha; };
```

Comment:
No need to mark `void` here, that's implicit.  For the other functions, I might consider marking the parameters const (like `const size_t N` instead of `size_t N`) even though it does nothing at all.  At least in mlpack, input options are generally denoted with a const specifier.  You can go either way here, since like I said it does nothing at all.